### PR TITLE
Update Ecobee documentation

### PR DESCRIPTION
**Pull request by hudcap**
## Proposed change

Add Ecobee documentation for services added in https://github.com/home-assistant/core/pull/38749

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/ecobee.markdown [hunk 1]
```diff
@@ -178,3 +181,31 @@ Sets the minimum amount of time that the fan will run per hour.
...
 | `entity_id`            | yes      | String or list of strings that point at `entity_id`'s of climate devices to control. Use `entity_id: all` to target all. |
 | `fan_min_on_time`      | no       | integer (e.g.,  5)                                                                                       |
+
+### Service `ecobee.set_dst_mode`
+
+Enable/disable automatic daylight savings time.
+
+| Service data attribute | Optional | Description                                       |
+| ---------------------- | -------- | ------------------------------------------------- |
+| `entity_id`            | no       | ecobee thermostat on which to set daylight savings time mode |
```
Comment (frenck) [inline source/_integrations/ecobee.markdown]: Entity ID is optional, as an area could be targeted instead.

```suggestion
| `entity_id`            | yes      | ecobee thermostat on which to set daylight savings time mode |
```

File: source/_integrations/ecobee.markdown [hunk 2]
```diff
@@ -178,3 +181,31 @@ Sets the minimum amount of time that the fan will run per hour.
...
+| `dst_enabled`          | no       | true or false                                     |
+
+### Service `ecobee.set_mic_mode`
+
+Enable/disable Alexa mic (only for ecobee 4).
+
+| Service data attribute | Optional | Description                                       |
+| ---------------------- | -------- | ------------------------------------------------- |
+| `entity_id`            | no       | ecobee thermostat on which to set the mic mode    |
```
Comment (frenck) [inline source/_integrations/ecobee.markdown]:
```suggestion
| `entity_id`            | yes      | ecobee thermostat on which to set the mic mode    |
```

File: source/_integrations/ecobee.markdown [hunk 3]
```diff
@@ -178,3 +181,31 @@ Sets the minimum amount of time that the fan will run per hour.
...
+| `mic_enabled`          | no       | true or false                                     |
+
+### Service `ecobee.set_occupancy_modes`
+
+Enable/disable Smart Home/Away and Follow Me modes.
+
+| Service data attribute | Optional | Description                                       |
+| ---------------------- | -------- | ------------------------------------------------- |
+| `entity_id`            | no       | ecobee thermostat on which to set occupancy modes |
```
Comment (frenck) [inline source/_integrations/ecobee.markdown]:
```suggestion
| `entity_id`            | yes      | ecobee thermostat on which to set occupancy modes |
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by hudcap** [comment]
Fixed, thanks!
For my own understanding, how would an area be targeted with these services?

**Comment by hudcap** [comment]
> For my own understanding, how would an area be targeted with these services?

Ah, using `area_id` -- never came across that before.
Should that be included in the docs? Or is it understood that `area_id` is always a drop-in for `entity_id`?

**Comment by frenck** [comment]
It is not always a drop in for `entity_id`, as not all services use the new registration methods.

`area_id` isn't documented, because at this point, it is only usable from the automation editor in the UI. Hopefully we can change that in the future. Nevertheless, if one would make an automation in the UI, parts of the saved automation can of course manually be reused (and reveals the area ID)

**Review by frenck — APPROVED** [review]
Thanks for updating and the PR! 👍

**Pull request merged**